src/zenml/zen_server/routers/auth_endpoints.py

```diff
@@ -522,6 +529,20 @@ def api_token(
             detail="Not authenticated.",
         )
 
+    if not token.device_id and not token.api_key_id:
+        config = server_config()
+
+        # If not authenticated with a device or a service account, then a
+        # short-lived generic API token is returned.
+        return generate_access_token(
```

Comment:
I don't understand this early return here: Why is the verifying whether the user has permissions to create a pipeline run after this check? Is it not possible to request a pipeline scoped token without a device or API key?

My understanding would be that we want to maybe do this permission check if `pipeline_id` or `schedule_id` are set? Even that's not perfect in case of an unlisted pipeline, but covers more cases than the current implementation?

Reply:
@schustmi , this endpoint is to be used in two distinct situations:

* from the dashboard, to generate 1 hour API tokens and display them to users (which is why we return early here)
* from the CLI, to generate these so called "workload" API tokens used by pipeline steps to connect to ZenML

The distinction is made by how the user is authenticated: if it's authenticated with an API key or through the web login (authorized device) flow, they are issued a "workload" API token (for which either a pipeline ID or schedule ID or both should theoretically be set, but I'll leave it to you to decide if we can also enforce that rule). If it's not an API key nor a device, it's a dashboard API token, in which case we issue a generic 1 hour token.

You can totally take this over and do whatever you like with this endpoint if you have better ideas for how we should make this available to users. My only concern is this: API tokens that are not tied to an API key nor an authorized device should be short lived, because they cannot be revoked in case of security incidents unless the entire user account is deactivated or deleted.